test/bluemix.test.js

```diff
@@ -37,193 +37,222 @@ function itSkipIf(flag) {
 }
 
 describe('loopback:bluemix generator', function() {
+  this.timeout(30 * 60 * 1000); // 30 minutes
   beforeEach(common.resetWorkspace);
+  beforeEach(function createSandbox(done) {
+    helpers.testDirectory(SANDBOX, done);
+  });
 
-  beforeEach(function(done) {
-    fs.ensureDir(SANDBOX, function() {
-      process.chdir(SANDBOX);
-      common.createDummyProject(SANDBOX, 'test-app', done);
-    });
+  beforeEach(function createProject(done) {
+    common.createDummyProject(SANDBOX, 'test-app', done);
   });
 
   afterEach(function(done) {
     process.chdir('/');
     rimraf(SANDBOX, done);
   });
 
-  it('should generate datasources.bluemix.js', function(done) {
-    var gen = givenBluemixGenerator('--force --bluemix');
-    helpers.mockPrompt(gen, {
-      enableManifest: false,
-    });
-    gen.run(function() {
-      ygAssert.file('./server/datasources.bluemix.js');
-      done();
-    });
+  it('should generate datasources.bluemix.js', function() {
```

Comment:
we don't need to pass `done` in the parameter or call done() inside?

Reply:
@dhmlau ah, don't need to. same as I put in comment https://github.com/strongloop/loopback-datasource-juggler/pull/1616#discussion_r209622203, mocha handles if a `Promise` returned.